src/components/VmsList/Pool.js

```diff
@@ -28,34 +27,17 @@ const Pool = ({ pool, icons, onStart }) => {
   const icon = icons.get(iconId)
 
   return (
-    <div className={`col-xs-12 col-sm-6 col-md-4 col-lg-3`}>
-      <div className='card-pf card-pf-view card-pf-view-select card-pf-view-single-select'>
-        <div>
-          <span className={sharedStyle['operating-system-label']}>{ osName }</span>
-        </div>
-        <div className='card-pf-body'>
-          <div className={`card-pf-top-element ${style['card-icon']}`}>
-            <Link to={`/pool/${pool.get('id')}`}>
-              <VmIcon icon={icon} className={style['card-pf-icon']}
-                missingIconClassName='fa fa-birthday-cake card-pf-icon-circle' />
-            </Link>
-          </div>
-
-          <h2 className={`card-pf-title text-center ${style['status-height']}`}>
-            <Link to={`/pool/${pool.get('id')}`} className={style['vm-detail-link']}>
-              <p className={`${style['vm-name']} ${style['crop']}`} title={pool.get('name')} data-toggle='tooltip' id={`${idPrefix}-status-name`}>
-                {pool.get('name')}
-              </p>
-            </Link>
-            <p className={`${style['vm-status']}`} id={`${idPrefix}-status`}>
-              <VmStatusIcon state={state} />&nbsp;{stateValue}
-            </p>
-          </h2>
-
-          <VmActions isOnCard vm={pool.get('vm')} onStart={onStart} pool={pool} />
-        </div>
-      </div>
-    </div>
+    <BaseCard idPrefix={idPrefix}>
+      <BaseCard.Header>
+        <span className={sharedStyle['operating-system-label']}>{ osName }</span>
+      </BaseCard.Header>
+      <BaseCard.Icon url={`/pool/${pool.get('id')}`} icon={icon} />
+      <BaseCard.Title idPrefix={idPrefix} url={`/pool/${pool.get('id')}`} name={pool.get('name')} />
+      <BaseCard.Status idPrefix={idPrefix}>
```

Comment:
I've seen patternfly-react and react-bootstrap code use a context conatiner to pass things down like id prefixes.  It may be worth looking into it and having nested contexts automatically add id postfixes depending on how the components are nested.  Just a thought.

Reply:
Nice idea, I'll look at it.